Use different user to create zarf package will have permission issue when deploy the package

The issue steps:
- In CICD, i use ubuntu os, `root` account to build zarf package `zarf-package-myapp-amd64-1.0.tar.zst`.   
Command is `zarf package create . -c`, it's output `zarf.yaml` just like this:
```yaml
- name: kubevirt-namespace
  required: true
  manifests:
  - name: kubevirt-namespace
    namespace: kubevirt
    files:
    - ../../../../../root/.zarf-cache/oci/dirs/03a3e106d7a419551e85e780e32fbd95023f264b562169f6f57ae3b62e20abcb/manifests/kubevirt-namespace-0.yaml
  actions:
    onRemove:
      after:
      - cmd: ./zarf tools kubectl delete ns kubevirt
```
- I install this package in my PC using root account, it's OK.
- Then, in another CICD, it use ubuntu os and `ubuntu` account to create upgrade package `zarf-package-myapp-amd64-1.0-differential-1.2.tar.zst`.  
Actually the  `kubevirt-namespace` not changed, keep same with first zarf package.  
Command is `zarf package create . --differential zarf-package-myapp-amd64-1.0.tar.zst -c`, it's output `zarf.yaml` like this:
```yaml
- name: kubevirt-namespace
  required: true
  manifests:
  - name: kubevirt-namespace
    namespace: kubevirt
    files:
    - ../../../../../home/ubuntu/.zarf-cache/oci/dirs/03a3e106d7a419551e85e780e32fbd95023f264b562169f6f57ae3b62e20abcb/manifests/kubevirt-namespace-0.yaml
  actions:
    onRemove:
      after:
      - cmd: ./zarf tools kubectl delete ns kubevirt
```
- Then when i download the upgrade zarf package, and install it using `dev` account with command `zarf package deploy zarf-package-myapp-amd64-1.0-differential-1.2.tar.zst -c`, it failed with error permission denied:
```bash
ERR failed to deploy package: unable to deploy component "kubevirt-namespace": unable to read manifest file /home/ubuntu/.zarf-cache/oci/dirs/03a3e106d7a419551e85e780e32fbd95023f264b562169f6f57ae3b62e20abcb/manifests/kubevirt-namespace-0.yaml: open /home/ubuntu/.zarf-cache/oci/dirs/03a3e106d7a419551e85e780e32fbd95023f264b562169f6f57ae3b62e20abcb/manifests/kubevirt-namespace-0.yaml: permission denied

[ERROR] Zarf package deployment failed.
```

Because the final user is `dev`, it's `zarf-cache` should be `/home/dev/.zarf-cache`, but it still used `/home/ubuntu` folder, and we don't have this account.

So i think this is the bug for `zarf package create` with `zarf package deploy` when using different account.

--- a/src/pkg/packager/deploy.go
+++ b/src/pkg/packager/deploy.go
@@ -628,12 +628,9 @@ func (d *deployer) installManifests(ctx context.Context, pkgLayout *layout.Packa
 	installedCharts := []state.InstalledChart{}
 	for _, manifest := range component.Manifests {
 		for idx := range manifest.Files {
+			manifest.Files[idx] = fmt.Sprintf("%s-%d.yaml", manifest.Name, idx)
 			if helpers.InvalidPath(filepath.Join(manifestDir, manifest.Files[idx])) {
-				// The path is likely invalid because of how we compose OCI components, add an index suffix to the filename
-				manifest.Files[idx] = fmt.Sprintf("%s-%d.yaml", manifest.Name, idx)
-				if helpers.InvalidPath(filepath.Join(manifestDir, manifest.Files[idx])) {
-					return installedCharts, fmt.Errorf("unable to find manifest file %s", manifest.Files[idx])
-				}
+				return installedCharts, fmt.Errorf("unable to find manifest file %s", manifest.Files[idx])
 			}
 			if manifest.IsTemplate() {
 				path := filepath.Join(manifestDir, manifest.Files[idx])

